groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -765,787 +1431,10708 @@ void Optional_DataImp<TYPE>::emplace(
+        // Return a reference providing non-modifiable access to the underlying
+        // 'TYPE' object.  Throws a 'bsl::bad_optional_access' if the
+        // 'optional' object is disengaged.
+
+#endif  //BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
+
+    const TYPE *operator->() const;
+        // Return a pointer providing non-modifiable access to the underlying
+        // 'TYPE' object.  The behaviour is undefined if the 'optional' object
+        // is disengaged.
+
+#ifdef BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
+
+    const TYPE&  operator*() const &;
+    const TYPE&& operator*() const &&;
+        // Return a reference providing non-modifiable access to the underlying
+        // 'TYPE' object.  The behavior is undefined if the 'optional' object
+        // is disengaged.
+
+#else
+
+    const TYPE& operator*() const;
+        // Return a reference providing non-modifiable access to the underlying
+        // 'TYPE' object.  The behavior is undefined if the 'optional' object
+        // is disengaged.
+
+#endif  // BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
+
+#ifdef BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
+
+    template <class ANY_TYPE>
+    TYPE value_or(BSLS_COMPILERFEATURES_FORWARD_REF(ANY_TYPE) rhs) const&;
+        // If 'this->has_value() == true', return a copy of the contained
+        // 'value_type' object.  Otherwise, return the value of specified 'rhs'
+        // converted to 'value_type'.
+
+#ifdef BSL_COMPILERFEATURES_GUARANTEED_COPY_ELISION
+    template <class ANY_TYPE>
+    TYPE value_or(bsl::allocator_arg_t,
+                  allocator_type,
+                  BSLS_COMPILERFEATURES_FORWARD_REF(ANY_TYPE)) const&;
+        // If 'this->has_value() == true', return a copy of the contained
+        // 'value_type' object created using the provided allocator.
+        // Otherwise, return the value of specified 'rhs' converted to
+        // 'value_type' and  use the specified 'allocator' for the returned
+        // object.
+
+#endif  //BSL_COMPILERFEATURES_GUARANTEED_COPY_ELISION
+#else
+    template <class ANY_TYPE>
+    TYPE value_or(BSLS_COMPILERFEATURES_FORWARD_REF(ANY_TYPE) rhs) const;
+        // If 'this->has_value() == true', return a copy of the contained
+        // 'value_type' object.  Otherwise, return the value of specified 'rhs'
+        // converted to 'value_type'.
+
+#ifdef BSL_COMPILERFEATURES_GUARANTEED_COPY_ELISION
+    template <class ANY_TYPE>
+    TYPE value_or(bsl::allocator_arg_t,
+                  allocator_type,
+                  BSLS_COMPILERFEATURES_FORWARD_REF(ANY_TYPE)) const;
+        // If 'this->has_value() == true', return a copy of the contained
+        // 'value_type' object created using the provided allocator.
+        // Otherwise, return the value of specified 'rhs' converted to
+        // 'value_type' and  use the specified 'allocator' for the returned
+        // object.
+
+#endif  //BSL_COMPILERFEATURES_GUARANTEED_COPY_ELISION
```

Comment:
Please make sure there's some consistency regarding these comments. Just having the macro being checked (with a space after `//`) should suffice.  I highlighted a couple places to fix, but there's more.
```suggestion
#endif  // BSL_COMPILERFEATURES_GUARANTEED_COPY_ELISION
```

Reply:
done as a part of a previous comment